assets/ts/schedule/components/SchedulePage.tsx

```diff
@@ -0,0 +1,446 @@
+    alerts,
+    variant: busVariantId
+  } = schedulePageData;
+  const { modalMode, modalOpen, selectedOrigin } = state;
+  const maybeStopTree: StopTree | null = stopTree
+    ? fromStopTreeData(stopTree)
+    : null;
+
+  return isFerryRoute(route) ? (
+    <>
+      <ScheduleFinder
+        updateURL={updateURL}
+        route={route}
+        stops={stops}
+        services={services}
+        routePatternsByDirection={routePatternsByDirection}
+        today={today}
+        modalMode={modalMode}
+        modalOpen={modalOpen}
+        directionId={directionId}
+        changeDirection={changeDirection}
+        selectedOrigin={selectedOrigin}
+        changeOrigin={changeOrigin}
+        closeModal={closeModal}
+        scheduleNote={scheduleNote}
+      />
+      <div className="schedule-map-container">
+        <h2>Route Map</h2>
+        {staticMapData && (
+          <>
+            <img
+              src={staticMapData.img_src}
+              alt={`${route.name} route map`}
+              className="img-fluid"
+            />
+            <a
+              href={staticMapData.pdf_url}
+              target="_blank"
+              rel="noopener noreferrer"
+            >
+              <i className="fa fa-search-plus" aria-hidden="true" />
+              View map as a PDF
+            </a>
+          </>
+        )}
+      </div>
+    </>
+  ) : (
+    <ScheduleDirection
+      directionId={directionId}
+      route={route}
+      routePatternsByDirection={routePatternsByDirection}
+      routeStopLists={routeStopLists}
+      mapData={mapData}
+      staticMapData={staticMapData}
+      stopTree={maybeStopTree}
+      alerts={alerts}
+      busVariantId={busVariantId}
+    />
+  );
+};
+
+const getScheduleFinder = (
+  schedulePageData: SchedulePageData,
+  state: StoreProps,
+  directionId: DirectionId
+): JSX.Element | undefined => {
+  const {
+    route,
+    stops,
+    services,
+    route_patterns: routePatternsByDirection,
+    today,
+    schedule_note: scheduleNote
+  } = schedulePageData;
+  const { modalMode, modalOpen, selectedOrigin } = state;
+  return (
+    <ScheduleFinder
+      updateURL={updateURL}
+      route={route}
+      stops={stops}
+      services={services}
+      routePatternsByDirection={routePatternsByDirection}
+      today={today}
+      modalMode={modalMode}
+      modalOpen={modalOpen}
+      directionId={directionId}
+      changeDirection={changeDirection}
+      selectedOrigin={selectedOrigin}
+      changeOrigin={changeOrigin}
+      closeModal={closeModal}
+      scheduleNote={scheduleNote}
+    />
+  );
+};
+
+const getScheduleNote = (
+  schedulePageData: SchedulePageData,
+  state: StoreProps,
+  directionId: DirectionId
+): JSX.Element => {
+  const {
+    pdfs,
+    hours,
+    route,
+    schedule_note: scheduleNote,
+    route_patterns: routePatternsByDirection,
+    today,
+    services,
+    stops
+  } = schedulePageData;
+  const { modalOpen, modalMode, selectedOrigin } = state;
```

Comment:
**question**: Would it be better to do `const state = useSelector((state: StoreProps) => state);` here instead of passing it in as a prop? Also applies to the `<ScheduleFinder />` component itself.